server/src/main/scala/viper/gobraserver/GobraServer.scala

```diff
@@ -51,50 +53,61 @@ object GobraServer extends GobraFrontend {
     _server.start()
     VerifierState.flushCachedDiagnostics()
   }
-  
 
-  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]) {
+  def restart() {
+    _server.stop()
+    ViperBackends.ViperServerBackend.resetServer()
+    init(_options)
+    start()
+  }
+
+  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]): Future[VerifierResult] = {
 
     val fileUri = fileData.fileUri
-    val filePath = fileData.filePath
 
-    resultFuture.onComplete {
-      case Success(_) => // ignore -> handled by reporter
+    // do some post processing if verification has failed
+    resultFuture.recoverWith({ case exception =>
+      // restart Gobra Server and then update client state
+      restart()
```

Comment:
Why is the restart required here? Gobra now does not crash anymore but sends this special message. The ViperServer should also not have been killed ... I might answer my own question, but is the problem that the Crash of Gobra might have been caused by a crash of the Viper verification backend, which also crashes the Viperserver? To summarize, is the problem that ViperServer might have crashed?

Reply:
I would need to try what happens without restart. Silas has already restarted everything and I only moved it before sending notifications to the client

Reply:
In the case of a failing error back translation, a restart is actually not necessary. However, I'm unable to estimate what would happen if Gobra crashes at a different place or a backend crashes